src/features/tasks/TaskSlice.ts

```diff
@@ -0,0 +1,38 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { fetchTasks } from "../../app/store/redusers/ActionCreators";
+import { ITask } from "./types";
+
+interface TaskState {
+  tasks: ITask[];
+  isLoading: boolean;
+  error: string;
+}
+
+const initialState: TaskState = {
+  tasks: [],
+  isLoading: false,
+  error: "",
+};
+
+export const taskSlice = createSlice({
+  name: "task",
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+   builder
+     .addCase(fetchTasks.pending, (state) => {
+       state.isLoading = true;
+     })
+     .addCase(fetchTasks.fulfilled, (state, action: PayloadAction<ITask[]>) => {
+       state.isLoading = false;
+       state.error = "";
+       state.tasks = action.payload;
+     })
+     .addCase(fetchTasks.rejected, (state, action) => {
+      state.isLoading = false;
+      state.error = action.payload as string || action.error.message || "Unknown error";
+     });
+ },
+});
+
```

Comment:
1. export const useTasksSelector = () => useAppSelector((state) => state.tasks)) // use this approach
2. const selectTasks = (state) => state.tasks

const {tasks, isLoading, error} = useTasksSelector()